core/templates/domain/statistics/learner-answer-details-backend-api.service.ts

```diff
@@ -16,52 +17,59 @@
  * @fileoverview Service to record learner answer info.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { downgradeInjectable } from '@angular/upgrade/static';
 
-require('domain/statistics/statistics-domain.constants.ajs.ts');
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { StatisticsDomainConstants } from
+  'domain/statistics/statistics-domain.constants';
 
-angular.module('oppia').factory('LearnerAnswerDetailsBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService', 'SUBMIT_LEARNER_ANSWER_DETAILS_URL',
-  function(
-      $http, $q, UrlInterpolationService, SUBMIT_LEARNER_ANSWER_DETAILS_URL) {
-    var _recordLearnerAnswerDetails = function(
-        expId, stateName, interactionId, answer, answerDetails,
-        successCallback, errorCallback) {
-      var recordLearnerAnswerDetailsUrl = (
-        UrlInterpolationService.interpolateUrl(
-          SUBMIT_LEARNER_ANSWER_DETAILS_URL, {
-            entity_type: 'exploration',
-            entity_id: expId
-          }));
+@Injectable({
+  providedIn: 'root'
+})
+export class LearnerAnswerDetailsBackendApiService {
+  constructor(
+        private http: HttpClient,
+        private urlInterpolationService: UrlInterpolationService
+  ) { }
 
-      var payload = {
-        state_name: stateName,
-        interaction_id: interactionId,
-        answer: answer,
-        answer_details: answerDetails
-      };
+  private _recordLearnerAnswerDetails(
```

Comment:
Delete this function and move its code into the public `recordLearnerAnswerDetails`.